File: pulse-kotlin/src/commonTest/kotlin/com/fluxchess/pulse/kotlin/engine/CastlingTypeTest.kt
/*
 * Copyright 2013-2024 Phokham Nonava
 *
 * Use of this source code is governed by the MIT license that can be
 * found in the LICENSE file.
 */

package com.fluxchess.pulse.kotlin.engine

import kotlin.test.Test
import kotlin.test.assertEquals
import kotlin.test.assertFalse
import kotlin.test.assertTrue

class CastlingTypeTest {
	@Test
	fun `castlingTypes array should be valid`() {
		for (castlingType in castlingTypes.indices) {
			assertEquals(castlingTypes[castlingType], castlingType)
		}
	}

	@Test
	fun `isValidCastlingType should return true if castlingType is valid false otherwise`() {
		assertTrue(isValidCastlingType(KINGSIDE))
		assertTrue(isValidCastlingType(QUEENSIDE))

		assertFalse(isValidCastlingType(NO_CASTLING_TYPE))
	}
}

File: pulse-go/internal/pulse/engine/castling_type_test.go
/*
 * Copyright 2013-2024 Phokham Nonava
 *
 * Use of this source code is governed by the MIT license that can be
 * found in the LICENSE file.
 */

package engine

import (
	"testing"
)

func TestCastlingType(t *testing.T) {
	t.Run("CastlingTypes array should be valid", func(t *testing.T) {
		for ct := range CastlingTypes {
			if CastlingTypes[ct] != ct {
				t.Errorf("wanted CastlingTypes at index %d to be %v, but got %v", ct, ct, CastlingTypes[ct])
			}
		}
	})
}

func TestCastlingType_IsValidCastlingType(t *testing.T) {
	tests := []struct {
		castlingType CastlingType
		want         bool
	}{
		{castlingType: Kingside, want: true},
		{castlingType: Queenside, want: true},
		{castlingType: NoCastlingType, want: false},
	}
	for _, tt := range tests {
		t.Run("IsValidCastlingType should return true if castling type is valid, false otherwise", func(t *testing.T) {
			valid := IsValidCastlingType(tt.castlingType)
			if valid != tt.want {
				t.Errorf("IsValidCastlingType(%d) = %v, want %v", tt.castlingType, valid, tt.want)
			}
		})
	}
}
